filters/apiusagemonitoring/filter.go

```diff
@@ -0,0 +1,126 @@
+package apiusagemonitoring
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/zalando/skipper/filters"
+	"net/http"
+	"time"
+)
+
+const (
+	metricCountAll  = "http_count"
+	metricCount500s = "http5xx_count"
+	metricCount400s = "http4xx_count"
+	metricCount300s = "http3xx_count"
+	metricCount200s = "http2xx_count"
+	metricCount100s = "http1xx_count"
+	metricLatency   = "latency"
+)
+
+const (
+	stateBagKeyPrefix = "filter." + Name + "."
+	stateBagKeyState  = stateBagKeyPrefix + "state"
+)
+
+// apiUsageMonitoringFilter implements filters.Filter interface and is the structure
+// created for every route invocation of the `apiUsageMonitoring` filter.
+type apiUsageMonitoringFilter struct {
+	Paths []*pathInfo
+}
+
+// apiUsageMonitoringFilterContext contains information about the metrics tracking
+// for one HTTP exchange (one routing). It serves to pass information from
+// the `Request` call to the `Response` call (stored in the context's `StateBag`).
+type apiUsageMonitoringFilterContext struct {
+	// Begin is the earliest point in time where the request is observed
+	Begin time.Time
+}
+
+func (f *apiUsageMonitoringFilter) Request(c filters.FilterContext) {
+	// Gathering information from the initial request for further metrics calculation
+	c.StateBag()[stateBagKeyState] = &apiUsageMonitoringFilterContext{
+		Begin: time.Now(),
+	}
+}
+
+func (f *apiUsageMonitoringFilter) Response(c filters.FilterContext) {
+	mfc, ok := c.StateBag()[stateBagKeyState].(*apiUsageMonitoringFilterContext)
+	if !ok {
```

Comment:
Left over that opens up an interesting question. Can this happen at all? Should we deny to monitor anything?

I would suggest to measure the counts and status code but leave out the latency.